lib/rubygems/test_case.rb

```diff
@@ -808,8 +808,7 @@ def new_default_spec(name, version, deps = nil, *files)
     spec.files = files
 
     lib_dir = File.join(@tempdir, "default_gems", "lib")
-    lib_dir.instance_variable_set(:@gem_prelude_index, lib_dir)
-    $LOAD_PATH.unshift(lib_dir)
+    $LOAD_PATH.push(lib_dir)
```

Comment:
Could you please elaborate, how these could be equivalent (possibly better in the commit message :angel:)? Because gems always have precedence over StdLib, so how could append be the same as prepend? This IMO gives precedence to StdLib stuff instead of the RubyGems test suite stuff. But I might missing something.

Reply:
Default gems normally live at the end of the default $LOAD_PATH. So adding them in the end makes more sense in my opinion. If we add them at the front, we need to hack into the internals of the $LOAD_PATH so that the result of `Gem.load_path_insert_index` is still correct.

In any case, this is just a test simplification independent from the fix in this PR. I can change it back if you prefer.

Reply:
>  This IMO gives precedence to StdLib stuff instead of the RubyGems test suite stuff

Yes, this is correct, but it doesn't change a thing, because everything needed from the rubygems code base should've been required by then, and because we don't use dummy gems with file names conflicting with rubygems files.

But if you feel this is worse or incorrect, I'm fine reverting this change. As I said, it's not an important part of this PR.

Reply:
I'll revert the change though so that this PR focuses more on the bug fix.

Reply:
Actually, I'm not really sure, it feels more realistic to me to add default test gems at the end of the $LOAD_PATH rather than at the front.

So I'm not sure if it's worth reverting.

Reply:
From my POV, it'd be much better if this was separate PR. As you can see, tweaking `$LOAD_PATH` is delicate thing, which might have unexpected results.